.github/workflows/rpm-build-and-test.yml

```diff
@@ -0,0 +1,688 @@
+                              env.CP_EL8_VM9_LABEL || 'ci_vm9' }}
+                    ;;
+                'el9')
+                    CHROOT_NAME="rocky+epel-9-x86_64"
+                    DISTRO_NAME="EL"
+                    DISTRO_NAME_UPPER="EL"
+                    DISTRO_NAME_LOWER="el"
+                    DISTRO_VERSION="9"
+                    DISTRO_VERSION_MAJOR="9"
+                    PROV_DISTRO_VERSION_MAJOR="8"
+                    OPENMPI="openmpi"
+                    LABEL=${{ env.CP_EL9_VM9_LABEL &&
+                              env.CP_EL9_VM9_LABEL || 'ci_vm9' }}
+                    ;;
+                'leap15')
+                    CHROOT_NAME="opensuse-leap-15.4-x86_64"
+                    DISTRO_NAME="Leap"
+                    DISTRO_NAME_UPPER="LEAP"
+                    DISTRO_NAME_LOWER="leap"
+                    DISTRO_VERSION="15.4"
+                    DISTRO_VERSION_MAJOR="15"
+                    OPENMPI="openmpi3"
+                    LABEL=${{ env.CP_LEAP15_VM9_LABEL &&
+                              env.CP_LEAP15_VM9_LABEL || 'ci_vm9' }}
+                    ;;
+            esac
+            echo "CHROOT_NAME=$CHROOT_NAME" >> $GITHUB_ENV
+            echo "DISTRO_NAME=$DISTRO_NAME" >> $GITHUB_ENV
+            echo "DISTRO_VERSION=$DISTRO_VERSION" >> $GITHUB_ENV
+            echo "DISTRO_WITH_VERSION=$DISTRO_NAME_LOWER$DISTRO_VERSION" >> $GITHUB_ENV
+            echo "BUILD_CHROOT=/var/lib/mock/$CHROOT_NAME-${{ github.run_id }}/" >> $GITHUB_ENV
+            echo "STAGE_NAME=Functional on $DISTRO_NAME $DISTRO_VERSION" >> $GITHUB_ENV
+            echo "STAGE_TAGS=$STAGE_TAGS" >> $GITHUB_ENV
+            echo "FTEST_ARG=$FTEST_ARG" >> $GITHUB_ENV
+            echo "DISTRO=${DISTRO_NAME_UPPER}_$DISTRO_VERSION_MAJOR" >> $GITHUB_ENV
+            echo -n "PROVISION_DISTRO=${DISTRO_NAME_UPPER}_" >> $GITHUB_ENV
+            echo "${PROV_DISTRO_VERSION_MAJOR:-$DISTRO_VERSION_MAJOR}" >> $GITHUB_ENV
+            echo -n "DAOS_STACK_${DISTRO_NAME_UPPER}_" >> $GITHUB_ENV
+            echo "${PROV_DISTRO_VERSION_MAJOR:-$DISTRO_VERSION_MAJOR}_LOCAL_REPO=not_used" >> \
+                 $GITHUB_ENV
+            echo "LABEL=$LABEL" >> $GITHUB_ENV
+            echo -n "INST_RPMS=daos-client daos-tests daos-server daos-serialize" >> $GITHUB_ENV
+            echo -n " daos-tests-internal" >> $GITHUB_ENV
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+          fetch-depth: 500
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Request and Provision a Cluster
+        timeout-minutes: 7200
+        uses: ./.github/actions/provision-cluster
+        with:
+          condition: env.CP_SKIP_FUNC_TEST-${{ env.DISTRO }} != 'true' && \
+                     env.CP_SKIP_FUNC_TEST != 'true'
+      - name: Run Test
+        timeout-minutes: 7200
+        if: env.CP_SKIP_FUNC_TEST-${{ env.DISTRO }} != 'true' && env.CP_SKIP_FUNC_TEST != 'true'
```

Comment:
Are we no longer skipping the stage if there are no tests matching the tags?  Currently launch.py will return a non-zero exit status if no tests are found for the stage, which presumably would mark it as failed.  We have a mode switch which could be used to change this behavior specific to CI, but wouldn't we still want to save the provisioning time?

Reply:
Ahhh.  I had forgotten that we do that in Jenkins currently and indeed, is a good feature.  Do you mind if it gets deferred to the next round of updates (in progress as I mentioned earlier)?